### PR TITLE
feat(datetime): `format()` options

### PR DESCRIPTION
**Pull request by babiabeo**
Fix: #4284

### CODE REVIEW

**Review by iuioiua — CHANGES_REQUESTED** [review]
Hey David, could we add the following to `datetime/format.ts` instead?
```ts
export interface FormatOptions {
  utc?: boolean;
}
```
We should also have accompanying documentation. WDYT?

File: datetime/format.ts [hunk 1]
```diff
@@ -18,13 +18,29 @@ import { DateTimeFormatter } from "./_common.ts";
...
+): string {
   const formatter = new DateTimeFormatter(formatString);
-  return formatter.format(date);
+  return formatter.format(
+    date,
+    options.utc ? { timeZone: "UTC" } : undefined,
+  );
+}
+
+/** The options used for formatting date in {@linkcode format}. */
```
Comment (iuioiua) [inline datetime/format.ts:42]:
```suggestion
/** Options for {@linkcode format}. */
```

**Review by iuioiua — APPROVED** [review]
LGTM! Thanks! I'll let Yoshiya review too.

**Review by kt3k — APPROVED** [review]
This addition makes sense to me. LGTM

**Pull request merged**